Source/EasyNetQ/Producer/ClientCommandDispatcherSingleton.cs

```diff
@@ -101,8 +101,8 @@ private void StartDispatcherThread(ConnectionConfiguration configuration)
                 {
                     try
                     {
-                        var channelAction = queue.Take(cancellation.Token);
-                        channelAction();
+                        if (queue.TryTake(out var channelAction, -1, cancellation.Token))
+                            channelAction();
```

Comment:
Usually I prefer to add {} around if blocks, especially in such a core code, to avoid misleading error in future code addition, if I'm not wrong is also a guide lines on .net core development repo.

Reply:
Yep, you are right, fixed